### ISSUE TITLE
Zero LM option not working

### ISSUE THREAD

**Issue by rajeevbaalwan** [facebookresearch/wav2letter#750]
As per documentation provided to use ZeroLM only thing i need to do  is pass --lm='' to the decoder.

But this is not working as decoder is still trying to load the LM.

Here is my decoder config 

--lexicon=/hdd1/nfshare_hulk/wav2vec/libri_exp/am/lexicon.txt
--lm=''
--am=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp/librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs/001_model_#hdd1#nfshare_hulk#wav2vec#libri_exp#lists#test.trans.h5.tsv.bin
--test=/hdd1/nfshare_hulk/wav2vec/libri_exp/lists/test.trans.h5.tsv
--wordscore=1
--beamsize=500
--beamthreshold=25
--silweight=-0.5
--nthread_decoder=1
--smearing=max
--show=true

and crashed decoder output is 

I0715 07:15:56.003154  5076 Decode.cpp:106] Gflags after parsing 
--flagfile=; --fromenv=; --tryfromenv=; --undefok=; --tab_completion_columns=80; --tab_completion_word=; --help=false; --helpfull=false; --helpmatch=; --helpon=; --helppackage=false; --helpshort=false; --helpxml=false; --version=false; --adambeta1=0.90000000000000002; --adambeta2=0.999; --am=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp/librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs/001_model_#hdd1#nfshare_hulk#wav2vec#libri_exp#lists#test.trans.h5.tsv.bin; --am_decoder_tr_dropout=0; --am_decoder_tr_layerdrop=0; --am_decoder_tr_layers=1; --arch=network.arch; --archdir=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp/librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs; --attention=content; --attentionthreshold=2147483647; --attnWindow=no; --attnconvchannel=0; --attnconvkernel=0; --attndim=0; --batchsize=4; --beamsize=2500; --beamsizetoken=250000; --beamthreshold=25; --blobdata=false; --channels=1; --criterion=asg; --critoptim=sgd; --datadir=; --dataorder=input; --decoderattnround=1; --decoderdropout=0; --decoderrnnlayer=1; --decodertype=wrd; --devwin=0; --emission_dir=; --emission_queue_size=3000; --enable_distributed=true; --encoderdim=0; --eosscore=0; --eostoken=false; --everstoredb=false; --fftcachesize=1; --filterbanks=40; --flagsfile=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp/librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs/decode.cfg; --framesizems=25; --framestridems=10; --gamma=1; --gumbeltemperature=1; --input=h5; --inputbinsize=100; --inputfeeding=false; --isbeamdump=false; --iter=535125; --itersave=false; --labelsmooth=0; --leftWindowSize=50; --lexicon=/hdd1/nfshare_hulk/wav2vec/libri_exp/am/lexicon.txt; --linlr=-1; --linlrcrit=-1; --linseg=1; --lm=''; --lm_memory=5000; --lm_vocab=; --lmtype=kenlm; --lmweight=0; --localnrmlleftctx=0; --localnrmlrightctx=0; --logadd=false; --lr=0.20000000000000001; --lr_decay=9223372036854775807; --lr_decay_step=9223372036854775807; --lrcosine=false; --lrcrit=0.002; --max_devices_per_node=8; --maxdecoderoutputlen=200; --maxgradnorm=1; --maxisz=9223372036854775807; --maxload=-1; --maxrate=10; --maxsil=50; --maxtsz=9223372036854775807; --maxword=-1; --melfloor=1; --memstepsize=10485760; --mfcc=false; --mfcccoeffs=13; --mfsc=false; --minisz=0; --minrate=3; --minsil=0; --mintsz=0; --momentum=0; --netoptim=sgd; --noresample=false; --nthread=4; --nthread_decoder=1; --nthread_decoder_am_forward=1; --numattnhead=8; --onorm=target; --optimepsilon=1e-08; --optimrho=0.90000000000000002; --outputbinsize=5; --pctteacherforcing=100; --pcttraineval=100; --pow=false; --pretrainWindow=0; --replabel=1; --reportiters=0; --rightWindowSize=50; --rndv_filepath=; --rundir=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp; --runname=librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs; --samplerate=16000; --sampletarget=0; --samplingstrategy=rand; --saug_fmaskf=27; --saug_fmaskn=2; --saug_start_update=-1; --saug_tmaskn=2; --saug_tmaskp=1; --saug_tmaskt=100; --sclite=/hdd1/nfshare_hulk/wav2vec/wav2vec_exp/librispeech_clean_asg_12Conv_prelu_lowreg_512kernel_trainlogs/sclite; --seed=0; --show=true; --showletters=false; --silscore=0; --smearing=max; --smoothingtemperature=1; --softwoffset=10; --softwrate=5; --softwstd=5; --sqnorm=true; --stepsize=9223372036854775807; --surround=|; --tag=; --target=tkn; --test=/hdd1/nfshare_hulk/wav2vec/libri_exp/lists/test.trans.h5.tsv; --tokens=tokens.txt; --tokensdir=/hdd1/nfshare_hulk/wav2vec/libri_exp/am; --train=/hdd1/nfshare_hulk/wav2vec/libri_exp/lists/train.trans.clean100.h5.tsv; --trainWithWindow=false; --transdiag=4; --unkscore=-inf; --use_memcache=false; --uselexicon=false; --usewordpiece=false; --valid=/hdd1/nfshare_hulk/wav2vec/libri_exp/lists/test.trans.h5.tsv; --warmup=1; --wav2vec=true; --wav2vecfeat=512; --weightdecay=0; --wordscore=0; --wordseparator=|; --world_rank=0; --world_size=1; --alsologtoemail=; --alsologtostderr=false; --colorlogtostderr=false; --drop_log_memory=true; --log_backtrace_at=; --log_dir=; --log_link=; --log_prefix=true; --logbuflevel=0; --logbufsecs=30; --logemaillevel=999; --logfile_mode=436; --logmailer=/bin/mail; --logtostderr=true; --max_log_size=1800; --minloglevel=0; --stderrthreshold=2; --stop_logging_if_full_disk=false; --symbolize_stacktrace=true; --v=0; --vmodule=; 
I0715 07:15:56.004304  5076 Decode.cpp:127] Number of classes (network): 29
I0715 07:15:56.289865  5076 Decode.cpp:134] Number of words: 89346
terminate called after throwing an instance of 'util::ErrnoException'
  what():  /root/kenlm/util/file.cc:76 in int util::OpenReadOrThrow(const char*) threw ErrnoException because `-1 == (ret = open(name, 00))'.
**No such file or directory while opening ''**
*** Aborted at 1594797356 (unix time) try "date -d @1594797356" if you are using GNU date ***
PC: @     0x7fd966923e97 gsignal
*** SIGABRT (@0x13d4) received by PID 5076 (TID 0x7fd9ac9ae380) from PID 5076; stack trace: ***
    @     0x7fd9a4cc6890 (unknown)
    @     0x7fd966923e97 gsignal
    @     0x7fd966925801 abort
    @     0x7fd967318957 (unknown)
    @     0x7fd96731eab6 (unknown)
    @     0x7fd96731eaf1 std::terminate()
    @     0x7fd96731ed24 __cxa_throw
    @     0x5578673196f3 util::OpenReadOrThrow()
    @     0x55786731833e lm::ngram::RecognizeBinary()
    @     0x5578672d95dc lm::ngram::LoadVirtual()
    @     0x5578671fe009 w2l::KenLM::KenLM()
    @     0x5578670b0450 main
    @     0x7fd966906b97 __libc_start_main
    @     0x55786710f37a _start
Aborted (core dumped)

Am i missing something ?

**Comment by lunixbochs** [comment]
ZeroLM worked for me, are you on an older version of wav2letter maybe?

**Comment by rajeevbaalwan** [comment]
I'm on latest commit of official wav2letter's master branch

This is the latest commit in my git log:

commit 18a1dfe4b22e875e5fd1f875f7bf9ec88a5616cd (upstream/master, origin/master, origin/HEAD, master)
Author: Shai Szulanski <iahs@fb.com>
Date:   Tue Jun 30 19:11:24 2020 -0700

    Run clang-format
    
    Summary:
    This directory is opted-in to clang-format but is not format-clean. This blocks continuous formatting from being enabled on fbcode, and causes hassle for other codemods that leave inconsistent formatting. Th
is diff runs clang-format, which is widely used and considered safe.
    
    If you are unhappy with the formatting of a particular block, please *accept this diff* and then in a stacked commit undo the change and wrap that code in `// clang-format off` and `// clang-format on`, or `
/* clang-format off */` and `/* clang-format on */`.
    
    drop-conflicts
    
    Reviewed By: an918tw
    
    Differential Revision: D22312482
    
    fbshipit-source-id: 99d77bd4416b70582bcea7966f73c444ecf5422e

**Comment by tlikhomanenko** [comment]
The problem in the expression you set: it shouldn't be lm='' - here `''` will be set as string for lm path. you need to put it empty like `--lm= --lexicon=...` or in the config file:
```
--lexicon=/hdd1/nfshare_hulk/wav2vec/libri_exp/am/lexicon.txt
--lm=
--wordscore=3
```

**Comment by rajeevbaalwan** [comment]
> The problem in the expression you set: it shouldn't be lm='' - here `''` will be set as string for lm path. you need to put it empty like `--lm= --lexicon=...` or in the config file:
> 
> ```
> --lexicon=/hdd1/nfshare_hulk/wav2vec/libri_exp/am/lexicon.txt
> --lm=
> --wordscore=3
> ```

will try this and let you know.
